File: 2nd_quarter/week01/Whaeun/추억 점수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    map<string, int> missing_score;
    
    for(int i = 0; i < name.size(); i++){
        missing_score[name[i]] = yearning[i];
    }
    
    for(int i = 0; i < photo.size(); i++){
        int sum = 0;
        for(int j = 0; j < photo[i].size(); j++){
            sum += missing_score[photo[i][j]];
        }
        answer.push_back(sum);
    }
    
    return answer;
}

File: 2nd_quarter/week01/Royce/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    arr = dict()
    
    for i in range(len(name)):
        arr[name[i]] = yearning[i]
    
    for i in range(0,len(photo)):
        num = 0
        for j in range(0,len(photo[i])):
            if photo[i][j] in arr.keys():
                photo[i][j] = arr[photo[i][j]]
            else:
                photo[i][j] = 0
            # 없을때는 0을 넣는 것도 고려해 주어야 한다.
            
    for i in photo:
        answer.append(sum(i))
        #     cnt = arr[photo[i][j]]
        #     num += cnt
        # answer.append(num)
    
    return answer
